src/backend/src/services/tasks.services.ts

```diff
@@ -258,4 +265,69 @@ export default class TasksService {
 
     return deletedTask.taskId;
   }
+
+  /**
+   * Sends the task deadline slack notifications for all tasks with a deadline of the given date
+   * @param deadline the deadline date as a number
+   */
+  static async sendTaskDeadlineSlackNotifications(deadline: number) {
+    const startOfDay = new Date(deadline);
+    const endOfDay = new Date(startOfDay);
+    endOfDay.setDate(startOfDay.getDate() + 1);
```

Comment:
what are u doing with all this date stuff? Kinda confused on what deadline is/ why you're passing it in as a number

Reply:
i had it like that just to reduce the number of new Date() in the controller, but i can change it to pass in a date instead. the number was just the date as a number in ms since epoch, which is what the new Date().setDate() returns